Fix demo project update build

**Description:**
After updating Demo project with the last change, one build has failed.

This is the text with the failure:
```text
2025-09-27T22:46:39.6267864Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ProcessPosix.cpp: In member function ‘void Process::_osInitilize()’:
2025-09-27T22:46:39.6276378Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ProcessPosix.cpp:41:29: warning: format ‘%lu’ expects argument of type ‘long unsigned int’, but argument 3 has type ‘id_type’ {aka ‘unsigned int’} [-Wformat=]
2025-09-27T22:46:39.6278815Z    41 |     sprintf( path, "/proc/%lu/cmdline", mProcessId );
2025-09-27T22:46:39.6279583Z       |                           ~~^           ~~~~~~~~~~
2025-09-27T22:46:39.6280056Z       |                             |           |
2025-09-27T22:46:39.6280531Z       |                             |           id_type {aka unsigned int}
2025-09-27T22:46:39.6281046Z       |                             long unsigned int
2025-09-27T22:46:39.6281446Z       |                           %u
2025-09-27T22:46:39.6283744Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ProcessPosix.cpp:45:33: warning: format ‘%lu’ expects argument of type ‘long unsigned int’, but argument 3 has type ‘id_type’ {aka ‘unsigned int’} [-Wformat=]
2025-09-27T22:46:39.6285884Z    45 |         sprintf( path, "/proc/%lu/exe", mProcessId );
2025-09-27T22:46:39.6286406Z       |                               ~~^       ~~~~~~~~~~
2025-09-27T22:46:39.6286866Z       |                                 |       |
2025-09-27T22:46:39.6287334Z       |                                 |       id_type {aka unsigned int}
2025-09-27T22:46:39.6287826Z       |                                 long unsigned int
2025-09-27T22:46:39.6288260Z       |                               %u
2025-09-27T22:46:39.6577102Z [ 23%] Building CXX object packages/areg-sdk-build/CMakeFiles/areg.dir/framework/areg/component/private/EventData.cpp.o
2025-09-27T22:46:39.9667765Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ThreadPosix.cpp: In static member function ‘static id_type Thread::_osGetCurrentThreadId()’:
2025-09-27T22:46:39.9671780Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ThreadPosix.cpp:122:13: error: invalid cast from type ‘pthread_t’ {aka ‘long unsigned int’} to type ‘id_type’ {aka ‘unsigned int’}
2025-09-27T22:46:39.9674416Z   122 |     return  reinterpret_cast<id_type>( pthread_self() );
2025-09-27T22:46:39.9675462Z       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-09-27T22:46:39.9706994Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ThreadPosix.cpp: In member function ‘Thread::eCompletionStatus Thread::_osDestroyThread(unsigned int)’:
2025-09-27T22:46:39.9710805Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ThreadPosix.cpp:129:27: error: invalid cast from type ‘id_type’ {aka ‘unsigned int’} to type ‘pthread_t’ {aka ‘long unsigned int’}
2025-09-27T22:46:39.9713218Z   129 |     pthread_t threadId  = reinterpret_cast<pthread_t>(Thread::INVALID_THREAD_ID);
2025-09-27T22:46:39.9714121Z       |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-09-27T22:46:39.9807011Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ThreadPosix.cpp:139:20: error: invalid cast from type ‘id_type’ {aka ‘unsigned int’} to type ‘pthread_t’ {aka ‘long unsigned int’}
2025-09-27T22:46:39.9809281Z   139 |         threadId = reinterpret_cast<pthread_t>(mThreadId);
2025-09-27T22:46:39.9836255Z       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-09-27T22:46:39.9838804Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ThreadPosix.cpp: In member function ‘bool Thread::_osCreateSystemThread()’:
2025-09-27T22:46:39.9841889Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ThreadPosix.cpp:183:35: error: invalid cast from type ‘pthread_t’ {aka ‘long unsigned int’} to type ‘id_type’ {aka ‘unsigned int’}
2025-09-27T22:46:39.9843879Z   183 |                 mThreadId       = reinterpret_cast<id_type>(handle->pthreadId);
2025-09-27T22:46:39.9844834Z       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-09-27T22:46:39.9847103Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ThreadPosix.cpp: In member function ‘Thread::eThreadPriority Thread::_osSetPriority(eThreadPriority)’:
2025-09-27T22:46:39.9850545Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/ThreadPosix.cpp:218:31: error: invalid cast from type ‘id_type’ {aka ‘unsigned int’} to type ‘pthread_t’ {aka ‘long unsigned int’}
2025-09-27T22:46:39.9852529Z   218 |         pthread_t threadId  = reinterpret_cast<pthread_t>(mThreadId);
2025-09-27T22:46:39.9853127Z       |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-09-27T22:46:39.9855401Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/SynchObjectsPosix.cpp: In member function ‘bool Mutex::_osLockMutex(unsigned int)’:
2025-09-27T22:46:39.9858736Z /home/runner/work/areg-sdk-demo/areg-sdk-demo/product/cache/gnu-x86-so/packages/areg-sdk-src/framework/areg/base/private/posix/SynchObjectsPosix.cpp:71:30: error: invalid cast from type ‘pthread_t’ {aka ‘long unsigned int’} to type ‘id_type’ {aka ‘unsigned int’}
2025-09-27T22:46:39.9860902Z    71 |         mOwnerThreadId.store(reinterpret_cast<id_type>(synchMutex->getOwningThreadId()));
2025-09-27T22:46:39.9861691Z       |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-09-27T22:46:40.3156109Z [ 23%] Building CXX object packages/areg-sdk-build/CMakeFiles/areg.dir/framework/areg/component/private/EventDataStream.cpp.o
2025-09-27T22:46:40.3999799Z gmake[2]: *** [packages/areg-sdk-build/CMakeFiles/areg.dir/build.make:1017: packages/areg-sdk-build/CMakeFiles/areg.dir/framework/areg/base/private/posix/ThreadPosix.cpp.o] Error 1
2025-09-27T22:46:40.4002035Z gmake[2]: *** Waiting for unfinished jobs....
2025-09-27T22:46:40.4687633Z gmake[2]: *** [packages/areg-sdk-build/CMakeFiles/areg.dir/build.make:1003: packages/areg-sdk-build/CMakeFiles/areg.dir/framework/areg/base/private/posix/SynchObjectsPosix.cpp.o] Error 1
2025-09-27T22:46:43.8932724Z gmake[1]: *** [CMakeFiles/Makefile2:1285: packages/areg-sdk-build/CMakeFiles/areg.dir/all] Error 2
2025-09-27T22:46:43.8934010Z gmake: *** [Makefile:136: all] Error 2
2025-09-27T22:46:43.8959065Z ##[error]Process completed with exit code 2.
2025-09-27T22:46:43.9106342Z Post job cleanup.
2025-09-27T22:46:44.0727698Z Post job cleanup.
2025-09-27T22:46:44.1654331Z [command]/usr/bin/git version
2025-09-27T22:46:44.1689545Z git version 2.51.0
```

The build must be fixed.

--- a/framework/areg/base/GETypes.h
+++ b/framework/areg/base/GETypes.h
@@ -81,7 +81,7 @@
         typedef unsigned long   ptr_type;
     #endif  // _UINTPTR_T_DEFINED
 
-#endif  // ptr_type
+#endif  // BIT64
 
 //! The sequence number type.
 typedef uint64_t        SequenceNumber;

--- a/framework/areg/base/NEUtilities.hpp
+++ b/framework/areg/base/NEUtilities.hpp
@@ -692,6 +692,34 @@ namespace   NEUtilities
     #pragma warning(default: 4251)
 #endif  // _MSC_VER
     };
+
+    /**
+     * \brief   Converts given value to pointer type ToPtr.
+     *          If ToPtr is not a pointer type, it will convert
+     *          the value to the given type.
+     * \tparam  ToPtr       The type to convert. If this is a pointer type,
+     *                      it will convert the value to pointer.
+     * \tparam  FromType    The type of value to convert.
+     * \param   value       The value to convert.
+     * \return  Returns converted value to pointer type ToPtr. If ToPtr is not
+     *          a pointer type, it will convert the value to the given type.
+     **/
+    template <typename ToPtr, typename FromType>
+    inline constexpr ToPtr convToPtr( FromType value );
+
+    /**
+     * \brief   Converts given pointer or value to numeric type ToNum.
+     *          If FromType is a pointer type, it will convert
+     *          the pointer to numeric value. Otherwise, it will
+     *          convert the value to the given type.
+     * \tparam  ToNum       The numeric type to convert.
+     * \tparam  FromType    The type of value to convert. If this is a pointer type,
+     *                      it will convert the pointer to numeric value.
+     * \param   ptrValue    The pointer or value to convert.
+     * \return  Returns converted pointer or value to numeric type ToNum.
+     **/
+    template <typename ToNum, typename FromType>
+    inline constexpr ToNum convToNum( FromType ptrValue );
 }
 
 //////////////////////////////////////////////////////////////////////////
@@ -769,6 +797,34 @@ inline uint64_t NEUtilities::Duration::passedMinutes( void ) const
 inline uint64_t NEUtilities::Duration::durationSinceStart( void ) const
 {
     return static_cast<uint64_t>((mStop > mStart ? (mStop - mStart).count( ) : (std::chrono::steady_clock::now( ) - mStart).count( )));
+}
+
+// ...existing code...
+
+template <typename ToPtr, typename FromType>
+inline constexpr ToPtr NEUtilities::convToPtr( FromType ptrValue )
+{
+    if constexpr (std::is_pointer_v<ToPtr>)
+    {
+        return reinterpret_cast<ToPtr>(static_cast<std::uintptr_t>(ptrValue));
+    }
+    else
+    {
+        return static_cast<ToPtr>(ptrValue);
+    }
+}
+
+template<typename ToNum, typename FromType>
+inline constexpr ToNum NEUtilities::convToNum(FromType value)
+{
+    if constexpr (std::is_pointer_v<FromType>)
+    {
+        return static_cast<ToNum>(reinterpret_cast<std::uintptr_t>(value));
+    }
+    else
+    {
+        return static_cast<ToNum>(value);
+    }
 }
 
 #endif  // AREG_BASE_NEUTILITIES_HPP

--- a/framework/areg/base/private/posix/ProcessPosix.cpp
+++ b/framework/areg/base/private/posix/ProcessPosix.cpp
@@ -29,6 +29,14 @@
 
 void Process::_osInitilize( void )
 {
+#ifdef BIT64
+    constexpr const char _fmtCmdLine[] = "/proc/%lu/cmdline";
+    constexpr const char _fmtExePath[] = "/proc/%lu/exe";
+#else   // defined(BIT32)
+    constexpr const char _fmtCmdLine[] = "/proc/%u/cmdline";
+    constexpr const char _fmtExePath[] = "/proc/%u/exe";
+#endif  // BIT64
+
     mProcessId = ::getpid( );
     mProcessHandle = static_cast<void *>(&mProcessId);
 
@@ -38,11 +46,11 @@ void Process::_osInitilize( void )
     ::memset( buffer, 0, File::MAXIMUM_PATH );
     ::memset( path, 0, 256 );
 
-    sprintf( path, "/proc/%lu/cmdline", mProcessId );
+    sprintf( path, _fmtCmdLine, mProcessId );
     FILE * file = ::fopen( path, "r" );
     if ( (file == nullptr) || (::fgets( buffer, File::MAXIMUM_PATH, file ) == nullptr))
     {
-        sprintf( path, "/proc/%lu/exe", mProcessId );
+        sprintf( path, _fmtExePath, mProcessId );
         ssize_t len = readlink( path, buffer, File::MAXIMUM_PATH );
         if ((len > 0) && (len < File::MAXIMUM_PATH))
         {

--- a/framework/areg/base/private/posix/SynchObjectsPosix.cpp
+++ b/framework/areg/base/private/posix/SynchObjectsPosix.cpp
@@ -19,6 +19,7 @@
 
 #include "areg/base/Thread.hpp"
 #include "areg/base/NEMemory.hpp"
+#include "areg/base/NEUtilities.hpp"
 #include "areg/base/DateTime.hpp"
 #include "areg/base/IESynchObject.hpp"
 #include "areg/base/private/posix/WaitableEventIX.hpp"
@@ -68,7 +69,7 @@ bool Mutex::_osLockMutex( unsigned int timeout )
 
     if ( NESynchTypesIX::SynchObject0 == SynchLockAndWaitIX::waitForSingleObject(*synchMutex, timeout) )
     {
-        mOwnerThreadId.store(reinterpret_cast<id_type>(synchMutex->getOwningThreadId()));
+        mOwnerThreadId.store(NEUtilities::convToNum<id_type, pthread_t>(synchMutex->getOwningThreadId()));
         result = true;
     }
 

--- a/framework/areg/base/private/posix/ThreadPosix.cpp
+++ b/framework/areg/base/private/posix/ThreadPosix.cpp
@@ -19,6 +19,7 @@
 #if defined(_POSIX) || defined(POSIX)
 
 #include "areg/base/IEThreadConsumer.hpp"
+#include "areg/base/NEUtilities.hpp"
 
 #ifndef _POSIX_PRIORITY_SCHEDULING
     #define _POSIX_PRIORITY_SCHEDULING
@@ -32,12 +33,12 @@
 #include <sys/types.h>
 
 #if __has_include(<sys/unistd.h>)
-#include <sys/signal.h>
-#include <sys/unistd.h>
-#else
-#include <unistd.h>
-#include <signal.h>
-#endif
+    #include <sys/signal.h>
+    #include <sys/unistd.h>
+#else   // !__has_include(<sys/unistd.h>)
+    #include <unistd.h>
+    #include <signal.h>
+#endif  // __has_include(<sys/unistd.h>)
 
 namespace
 {
@@ -125,14 +126,14 @@ void Thread::_osSleep(unsigned int timeout)
 
 id_type Thread::_osGetCurrentThreadId( void )
 {
-    return  reinterpret_cast<id_type>( pthread_self() );
+    return NEUtilities::convToNum<id_type, pthread_t>(pthread_self());
 }
 
 Thread::eCompletionStatus Thread::_osDestroyThread(unsigned int waitForStopMs)
 {
     // Initially, the thread is not valid and not running, nothing to destroy
     Thread::eCompletionStatus result = Thread::eCompletionStatus::ThreadInvalid;
-    pthread_t threadId  = reinterpret_cast<pthread_t>(Thread::INVALID_THREAD_ID);
+    pthread_t threadId  = NEUtilities::convToPtr<pthread_t, id_type>(Thread::INVALID_THREAD_ID);
 
     do
     {
@@ -142,7 +143,7 @@ Thread::eCompletionStatus Thread::_osDestroyThread(unsigned int waitForStopMs)
             return Thread::eCompletionStatus::ThreadInvalid;
         }
 
-        threadId = reinterpret_cast<pthread_t>(mThreadId);
+        threadId = NEUtilities::convToPtr<pthread_t, id_type>(mThreadId);
         _unregisterThread();
 
     } while(false);
@@ -186,7 +187,7 @@ bool Thread::_osCreateSystemThread( void )
             {
                 result          = true;
                 mThreadHandle   = static_cast<THREADHANDLE>(handle);
-                mThreadId       = reinterpret_cast<id_type>(handle->pthreadId);
+                mThreadId       = NEUtilities::convToNum<id_type, pthread_t>(handle->pthreadId);
                 mThreadPriority = Thread::eThreadPriority::PriorityNormal;
 
                 if (_registerThread() == false)
@@ -221,7 +222,7 @@ Thread::eThreadPriority Thread::_osSetPriority( eThreadPriority newPriority )
     if (_isValidNoLock() && (newPriority != oldPrio))
     {
         int schedPrio       { MIN_INT_32 };
-        pthread_t threadId  = reinterpret_cast<pthread_t>(mThreadId);
+        pthread_t threadId  { NEUtilities::convToPtr<pthread_t, id_type>(mThreadId) };
         switch (newPriority)
         {
         case Thread::eThreadPriority::PriorityLowest:

